service/src/main/java/bio/terra/externalcreds/services/ProviderService.java

```diff
@@ -184,30 +186,39 @@ public void deleteLink(String userId, String providerId) {
     linkedAccountService.deleteLinkedAccount(userId, providerId);
   }
 
-  public int validatePassportsWithAccessTokenVisas() {
-    var passportDetailsList = passportService.getPassportsWithUnvalidatedAccessTokenVisas();
-
-    passportDetailsList.forEach(
-        pd -> {
-          var responseBody = validatePassportWithProvider(pd);
-
-          // If the response is not "valid", get a new passport.
-          if (!responseBody.equalsIgnoreCase("valid")) {
-            log.info(
-                String.format(
-                    "Found a visa that is not valid, with validation response '%s'. Refreshing passport.",
-                    responseBody));
-            var linkedAccount = linkedAccountService.getLinkedAccount(pd.getLinkedAccountId());
-            try {
-              linkedAccount.ifPresentOrElse(
-                  this::authAndRefreshPassport,
-                  () -> log.info("No linked account found when trying to validate passport."));
-            } catch (Exception e) {
-              log.info("Failed to refresh passport, will try again at the next interval.", e);
-            }
+  public int validateAccessTokenVisas() {
+    var visaDetailsList = passportService.getUnvalidatedAccessTokenVisaDetails();
+
+    var linkedAccountIdsToRefresh =
+        visaDetailsList.stream()
+            .flatMap(
+                visaDetails -> {
+                  var responseBody = validateVisaWithProvider(visaDetails);
+                  log.info(
+                      "Got visa validation response: {}",
+                      Map.of(
+                          "linkedAccountId", visaDetails.getLinkedAccountId(),
+                          "providerId", visaDetails.getProviderId(),
+                          "validationResponse", responseBody));
```

Comment:
Yes, this is the log message I had in mind but you should not have the `: {}` at the end. Also, I would put this log statement in `validateVisaWithProvider`, then you can inline `responseBody` and this block of code gets much cleaner.